### ISSUE TITLE
Problem: Cross environment issues on Windows

### ISSUE THREAD

**Issue by gdevdeiv** [Blockrazor/blockrazor#1364]
Problem: `export` doesn't work on Windows, and so the Getting started guide is not valid for that OS. Maybe even could bring issues in the future (building the project on that arch).

Solution: use `cross-env` before the npm scripts, so the `ROOT_URL` and the `HTTP_FORWARDED_COUNT` flags can be set through that package and not through the bash command export.

**Comment by gdevdeiv** [comment]
I guess this is low priority, but just had the issue right now as I won't have my usual Mac machine around ;)

**Comment by AshHimself** [comment]
You can update the README to close this or maybe even see what @janat08 does as I believe he is on windows as well?

**Comment by gazhayes** [comment]
> export doesn't work on Windows, and so the Getting started guide is not valid for that OS. 

This isn't actually low priority - this is probably a relatively annoying barrier to entry for people unfortunate enough to be using Windows and the C4 works best when there minimal barriers to entry (e.g. if someone on Windows wants to fix one line of code once and never come back again they shouldn't need to spend an hour trying to get the thing to build).

> use cross-env before the npm scripts, so the ROOT_URL and the HTTP_FORWARDED_COUNT flags can be set through that package

The only potential issue I see with this is that the env variables are different for production and local dev so that would have to be handled somehow.

**Comment by gdevdeiv** [comment]
Proposed solution: Instead of using `meteor` and `meteor --production` manually to run the project, I would setup some pre-made one-line scripts in the package.json file. That way, by just calling `npm start`/`npm run dev` or `npm run build`/`npm run prod`, everything will be ready for you ;)

**Comment by janat08** [comment]
I quickly skipped those steps.

**Comment by gazhayes** [comment]
Cool :) don't forget to also update the relevant sections of the Readme if you do this one.

**Comment by gazhayes** [comment]
> I quickly skipped those steps.

@janat08 Did you have any problems?

**Comment by janat08** [comment]
Images upload doesn't work.

**Comment by gazhayes** [comment]
Oh this wouldn't fix that anyway, there are some hardcoded global variables for that somewhere.

**Comment by gazhayes** [comment]
It's not really a problem that image uploading doesn't work in dev environments, anyone working on image uploading related issues can see in the codebase how to make it work (it's fairly well commented I believed). And for people working on things were image uploading is required to pass validation there are simple temporary workarounds.

**Comment by gdevdeiv** [comment]
Then I'll take on this Issue just for the the `ROOT_URL` and the `HTTP_FORWARDED_COUNT` flags, I would however take a look at image uploading on dev environments even if it is "low prio", as it is a must for example when creating a coin (you can't do that using the wizard locally, without messing with validation).

@blockrazorbot claim.

**Comment by anbud** [comment]
I'm also on Windows, I haven't had any issue with these environmental variables, I've skipped it, and everything is working fine.

As for the image upload, create C:\var\www\static\images directory, and it'll start working  correctly. Thumbnails aren't working, but the code isn't failing anywhere.

**Comment by gdevdeiv** [comment]
Yes, I'm also skipping that and it works just fine, I will assume that the default values for each one are the ones used in development; and that in production we are using the "good values" for each of them. That is OK I guess, we could leave it there and nothing will happen. However it's not really documented, and more env variables could be used in the future. Maybe we could document that in the README.md file, and we even might implement DotEnv in the future, so a .env file could be used to store the env variables.

As for the image upload, I guess it's fine too, but the exact same thoughts are coming to me, we could use an environment variable to specify the UPLOAD_PATH, that would be better.

**Comment by gdevdeiv** [comment]
@blockrazorbot remove "in progress"
@blockrazorbot label "done"
